### PR TITLE
feat: refactor jmeterd executor to add better testing and support for better exit status handling

### PR DESCRIPTION
**Pull request by dejanzele**
## Pull request description 

Refactor executor so components are easily testable.
Add more unit tests and integration tests.
Add support for handling the scenario when test has an error and JMeter returns 0 exit code
Support for running regular JMeter if SLAVES_COUNT is omitted or set to 0 or 1.

This should close https://github.com/kubeshop/testkube/issues/4587 and https://github.com/kubeshop/testkube/issues/4589

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: contrib/executor/jmeterd/pkg/parser/csvjtl.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+				dict[header[i]] = record[i]
+			}
+			rows = append(rows, dict)
+		}
+	}
+	return rows, nil
+}
+
+func MapElementToResult(in map[string]string) CSVResult {
+	elapsed, _ := strconv.Atoi(in["elapsed"])
```
Comment (vsukhin) [inline contrib/executor/jmeterd/pkg/parser/csvjtl.go:71]: do we need to log error?

**Review by vsukhin — APPROVED** [review]
looks good, fix  unit tests

File: config/slave-pod-template.yml [hunk 2]
```diff
@@ -3,11 +3,13 @@ kind: Pod
 metadata:
   name: "{{ .Name }}"
   namespace: {{ .Namespace }}
+  {{- if .JobUID }}
```
Comment (vsukhin) [inline config/slave-pod-template.yml:6]: don't forget tto change it in helm-chart

File: contrib/executor/jmeterd/pkg/parser/csvjtl.go (hunk 1 above)
Comment (dejanzele) [reply contrib/executor/jmeterd/pkg/parser/csvjtl.go:71]: Good question, we parse the JTL file here, it shouldn't ever error, unless JMeter changes it to something else in future versions which doesn't seem likely.
Even if it failed, I don't know should we fail the test, but might make sense to log the error at least

File: contrib/executor/jmeterd/pkg/runner/runner.go [hunk 3]
```diff
@@ -148,146 +129,116 @@ func (r *JMeterDRunner) Run(ctx context.Context, execution testkube.Execution) (
...
 	entryPoint := getEntryPoint()
 	for i := range execution.Command {
 		if execution.Command[i] == "<entryPoint>" {
 			execution.Command[i] = entryPoint
+			break
 		}
 	}
 
+	d, _ := exec.Command("pwd").Output()
+	fmt.Println(string(d))
```
[the remark this answers is not in the document]
Comment (dejanzele) [reply contrib/executor/jmeterd/pkg/runner/runner.go]: yep, I removed it

**Pull request merged**